### PR TITLE
Improve error handling in image-upload when uploading a DV with already existing PVC

### PR DESCRIPTION
**Pull request by alromeros**
**What this PR does / why we need it**:

This Pull Request aims to improve the error handling in image-upload so, when uploading a DV with an already existing PVC (PVC with the same name already exists), we trigger a different error message depending on if the PVC is related or not to the DV.

1. If the PVC contains the `DeleteAfterCompletion` annotation, we assume it was created and populated by CDI and that its DV has been already garbage-collected. We now reflect that in the error message.
2. If the PVC doesn't contain that annotation, we assume it's independent of any DV. We trigger the already existing error message.

**Which issue(s) this PR fixes**:
Fixes # https://bugzilla.redhat.com/show_bug.cgi?id=2176422

**Special notes for your reviewer**:

I think we can't simply change the current error message to match the expected result in the [bz](https://bugzilla.redhat.com/show_bug.cgi?id=2176422) since we don't know if the PVC exists due to an already garbage-collected DV or if it was just created with the same name as the uploaded DV.

Instead of just changing that error msg, I'm checking if the PVC contains the `DeleteAfterCompletion` annotation to determine if it was created by a DV. I think it's a good solution but I'm not 100% happy with that because the PVC could have been created by a completely different DV with the same name. Would appreciate some comments with this in mind.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/virtctl/imageupload/imageupload.go [hunk 1]
```diff
@@ -709,7 +710,14 @@ func getAndValidateUploadPVC(client kubecli.KubevirtClient, namespace, name stri
 	if !createPVC {
 		_, err = client.CdiClient().CdiV1beta1().DataVolumes(namespace).Get(context.Background(), name, metav1.GetOptions{})
 		if err != nil {
+			// When the PVC exists but the DV doesn't, there are two possible outcomes:
 			if k8serrors.IsNotFound(err) {
+				// 1. The DV was already garbage-collected. The PVC was created and populated by CDI as expected.
+				dvGarbageCollected := pvc.Annotations[deleteAfterCompletionAnnotation]
```
Comment (ShellyKa13) [inline pkg/virtctl/imageupload/imageupload.go:716]: I wonder if maybe we should check the PodPhaseAnnotation like done after this if and thats how we will know the PVC is already populated?

File: pkg/virtctl/imageupload/imageupload.go [hunk 2]
```diff
@@ -709,7 +710,14 @@ func getAndValidateUploadPVC(client kubecli.KubevirtClient, namespace, name stri
...
+				if dvGarbageCollected == "true" {
```
Comment (awels) [inline pkg/virtctl/imageupload/imageupload.go:717]: You can also do
```go
if pvc.Anntations[deleteAfterCompletionAnnotation] == "true" {
...
}
```

File: pkg/virtctl/imageupload/imageupload.go (hunk 1 above)
Comment (arnongilboa) [reply pkg/virtctl/imageupload/imageupload.go:716]: PVC can be populated not only by CDI, so this is just a hint.

File: pkg/virtctl/imageupload/imageupload.go (hunk 2 above)
Comment (alromeros) [reply pkg/virtctl/imageupload/imageupload.go]: Considering the previous comment I'll do something like:
```go
if dvGarbageCollected := pvc.Annotations[deleteAfterCompletionAnnotation] == "true" &&
    pvc.Annotations[PodPhaseAnnotation] == string(v1.PodSucceeded); dvGarbageCollected {
```
wdyt?

**Pull request merged**